main.go

```diff
@@ -2,16 +2,30 @@ package main
 
 import (
 	"./logging"
+	"fmt"
 	"log"
 	"os"
 	"strings"
+	"sync"
+
+	"github.com/hashicorp/go-reap"
 )
 
 func main() {
 	log.SetOutput(os.Stdout)
 	if strings.EqualFold(os.Getenv("DEBUG"), "true") {
 		go logging.StartLogging()
 	}
+	// Get feedback on reaped children and errors.
+	if reap.IsSupported() {
+		pids := make(reap.PidCh, 2)
+		errors := make(reap.ErrorCh, 2)
+		done := make(chan struct{})
+		var reapLock sync.RWMutex
+		go reap.ReapChildren(pids, errors, done, &reapLock)
```

Comment:
I haven't used this library myself. Judging by a quick look at the project description, this seems to be the solution.

I would add integration tests for this functionality (skip unit testing altogether). We can, for example:

* Run the `proxy` and `go-demo` (both are used in the current integration tests)
* send several `go-demo` `reconfigure` requests so that zombie processes are generated
* enter into the container and confirm that zombie processes are gone. This should be executed in a loop with a timeout in case the library needs some time to reap the processes.
* send a request to `go-demo` through the `proxy` and confirm that everything still works.

What do you think? Integration tests are located in `integration_tests/integration_tests_swarm.go`. Would you like to try adding them? If you have trouble with the tests, I can do it as well.

It would be nice if unit tests are added as well but I cannot think what would be a good scenario that would not make the whole suite slow. Currently it takes around a second or two to run all UTs.